src/ssb_project_cli/ssb_project/build/poetry.py

```diff
@@ -101,6 +102,37 @@ def poetry_source_add(
         cwd=cwd,
     )
 
+    # If the lock is created off-prem, we need to refresh the lock.
+    if nexus_source_not_set_in_lock(source_url, cwd):
+        print("Refreshing lock file...")
+        execute_command(
+            "poetry lock --no-update".split(" "),
+            "source-remove",
+            "Poetry successfully refreshed lock file!",
+            "Poetry failed to refresh lock file.",
+            cwd=cwd,
+        )
+
+
+def nexus_source_not_set_in_lock(source_url: str, cwd: Path) -> bool:
```

Comment:
This was hard to read. I think it is better to call the method something like `exists_in_lockfile` and then use the `not` operator in the if-check. Something like:  `if not exists_in_lockfile(source_url, cwd)`

Reply:
The problem with that name(exists_in_lockfile) is that i also want it to return false when the file does not exist.

If the file the lockfile is not created, the source is not set there.

Reply:
Is "is_source_not_set_in_lock_file" better?

Reply:
I think `should_update_lock_file` is less confusing.

Reply:
should_update_lock_file works 👍